python/lsst/ts/mthexapod/hexapod_csc.py

```diff
@@ -429,8 +448,8 @@ async def compensation_loop(self) -> None:
                     is_compensation_loop=True,
                 )
             except Exception:
-                self.log.exception("Compensation failed; turning off compensation mode")
-                await self.evt_compensationMode.set_write(enabled=False)
+                self.log.exception("Compensation failed; CSC going to Fault.")
+                await self.fault(-2, report="Compensation failed.")
```

Comment:
yeah, but I can't push any xml changes to the summit now. I will make a ticket to update this later.

Reply:
Just notice one thing here.
Since this fault comes from the CSC, the controller would be still under the Enabled state, which means the Copley drive is still on with the current.
Therefore, please consider to put the controller back to the standby state first to disable the Copley drive before faulting the CSC.

The followings are the possible scenarios:

1. Controller is in the Fault state. CSC should be in Fault as well. If the CSC is not in fault yet, controller has disabled the Copley drive already in fault. The Copley should be safe.
2. Controller is in the Standby state. This should not happen otherwise the CSC could not be enabled.
3. Controller is in the Enabled/Stationary state. CSC can issue the standby command to controller directly.
4. Controller is in the Enabled/Pt2Pt state. CSC needs to issue a stop command first. After the controller stops, the state will be Enabled/Stationary state. Then, CSC can issue the standby command.

Thanks!

Reply:
Another possibility is that the controller provides a command/interface to CSC to let it be able to fail the controller. Then, this fail will disable the Copley drive as well to protect the drives.

Reply:
I think the simplest method would be:

```python
try:
    await self.stop_motion()
except Exception:
    pass

// BaseCSC.begin_disable() in ts_hexrotcomm
// the following code might need to be modified.
// This is to standby the controller and turn off the drives
try:
    await self.run_command(
        code=self.CommandCode.SET_STATE,  # type: ignore[attr-defined]
        param1=SetStateParam.STANDBY,
    )
    await self._enable_drives(False)
except Exception:
    pass
```